polaris-react/src/components/Badge/Badge.scss

```diff
@@ -11,29 +13,59 @@
   border-radius: var(--p-border-radius-5);
   color: var(--p-color-text);
 
+  :where(#{$se23}) & {
+    // stylelint-disable-next-line -- se23 bg-transparent-hover
+    background-color: rgba(26, 25, 26, 0.05);
+    // stylelint-disable-next-line -- se23 text-subdued
+    color: #616161;
+    // stylelint-disable-next-line -- se23 border radius 10px
+    border-radius: 10px;
+  }
+
   @media print {
     border: solid var(--p-border-width-1) var(--p-color-border);
   }
 }
 
 .statusSuccess {
   background-color: var(--p-color-bg-success);
+
+  #{$se23} & {
+    color: var(--p-color-text-success);
+  }
 }
 
 .statusInfo {
   background-color: var(--p-color-bg-info);
+
+  #{$se23} & {
+    color: var(--p-color-text-info);
+  }
 }
 
 .statusAttention {
   background-color: var(--p-color-bg-caution);
+
+  #{$se23} & {
+    color: var(--p-color-text-caution);
+  }
 }
 
 .statusWarning {
   background-color: var(--p-color-bg-warning);
+
+  #{$se23} & {
+    // stylelint-disable-next-line -- se23 color-text-warning
+    color: #412d00;
```

Comment:
you can add new colors to ``polaris-tokens/src/token-groups/color.ts`! for this I think it would be

```
  'color-text-warning': {
    value: ',
    valueExperimental: colorsExperimental.orange[15],
    description: '',
  },
```

and then add the key to `ColorTextAlias`